### ISSUE TITLE
[VMWare] Patch works fine, but game "launches" and immediately crashes with any release version

### ISSUE THREAD

**Issue by wkruddy** [ferib/D2R-Offline#76]
Nice work, I'm happy you were able to do this and get lots of people playing! No dice for me, however.

I usually run cracked things in a VM, for obvious reasons. I've followed all the steps multiple times but can't actually get the game to boot up.

I read some other comment that it might be related to minimum system reqs? I'm running on a 6800XT and 64GB of RAM on the main machine... and then giving my VM 16GB of RAM, and it has 50GB of extra space, etc. I'm worried it's a graphical issue since it's using virtualized hardware...

Do you have any insight on what might be going on, or is there a way to open it in `verbose` mode so I can do some debugging and/or a way to read crash logs? I presume you've seen people talk about it crashing immediately, and I know I've seen other people comment about it, without a great solution. Thanks!

**Comment by mfrklic** [comment]
I would doubt the game would run on virtualized hardware mainly because of the 3D engine and graphics cards requirements. You can test the game by itself just running game.exe and see if you get the same results.

I understand the concern of running cracked software which is why the complete source code is here and can be compiled yourself. While I do not 100% know much about the actual in memory patching process portion, if there was any nefarious code it would have been caught by those with more knowledge that are out there using it themselves.

**Comment by wkruddy** [comment]
Yeah, I was worried that would be the problem. The actual `Game.exe` also crashes the same way. I can run games like GuildWars2 on max graphics in a VM using the virtualized hardware, so it's a bummer I can't seem to get this one to play.

It's just good OpSec to run it on a VM, ya know? Even with the code available, because AV still scans it as a virus every time it sees it; albeit, a different virus every time so it's likely a red herring. Curious if there's a way to log the failures though.

**Comment by mfrklic** [comment]
Understand completely. The D2R game code is still alpha so probably not optimized to support things like this either. I would think the game may still produce some kind of crash logs somewhere but I am not sure off the top of my head. I recall classic may have created crash logs in the game folder but someone with more experience with that hopefully can chime in.

**Comment by creambyemute** [comment]
@wkruddy If the game.exe itself crashes it's a problem with the game and not the crack/patcher as far as I know.

**Comment by Stupreme** [comment]
> Yeah, I was worried that would be the problem. The actual `Game.exe` also crashes the same way. I can run games like GuildWars2 on max graphics in a VM using the virtualized hardware, so it's a bummer I can't seem to get this one to play.
> 
> It's just good OpSec to run it on a VM, ya know? Even with the code available, because AV still scans it as a virus every time it sees it; albeit, a different virus every time so it's likely a red herring. Curious if there's a way to log the failures though.

Running games in a VM is never good. If you are worried of malware from an OpSec standpoint then it would be better to honeypot potential malware and analyze the behavior. 

In this case you can simply download the code, read through it, and compile it yourself.

**Comment by gandalfthewhite19890404** [comment]
Game "works"(just closes with good result in console) same way in sandbox environment (in Sandboxie), maybe it is a some anticheat system.
So it could be not a D2ROffline issue but game issue.

**Comment by malvarenga123** [comment]
Does the game require DX12? The same thing described here happens on Windows 7. Game.exe loads and closes immediately. There is no log to see what is causing this.

**Comment by mfrklic** [comment]
The official D2R system requirements say you need Windows 10:

https://us.battle.net/support/en/article/284798

**Comment by malvarenga123** [comment]
I know, but mainly the only thing that would stop the game from running on Windows 7 would be DirectX 12.
Since this is happening on virtual machines with Windows 10, the error might be elsewhere, but an error log is needed.

**Comment by ferib** [comment]
Explain the 'crash', black screen? no screen at all? error message?

**Comment by malvarenga123** [comment]
Yes, when Game.exe loads, a black screen (windowed full screen) loads and closes immediately. No error message, log, or anything.
As others have said, I don't think it's a problem with the loader, but with D2R itself.

**Comment by ferib** [comment]
@malvarenga123 Can you check if there are any `game.exe`'s running in task manager? I know the game does this behavior when it detects an instance of the game is already created. There are probably multiple things that can cause this kind of 'crash', but this is the only one I can think of atm.

**Comment by malvarenga123** [comment]
It's not, you can barely see it in the task manager because it closes really fast.

**Comment by ferib** [comment]
Oh okay nvm, if you only see one game process you good, I assume its hardware issues then,

**Comment by mfrklic** [comment]
@malvarenga123 Unfortunately as far as I can see there is no error log. So it's kind of hit or miss. Some people were able to get passed issues like this by updating graphics card drivers.

**Comment by Chris-HC1** [comment]
I had the same issue in Oracle VM (Win 10) running on Win7 machine
There was nearly no difference between my starting attemts in Win10 VM or Win 7. I really tried a lot of different starting parameters, driver updates, antivirus program,premissions: But: Game.exe launches ( thru D2ROffline.exe in windowed full screen) =>crashes immediately. Blizzard error logs says: Access voilation with not much usable information.
Nevertheless when i did a clean Win10 install on an external drive and used the same game files to start it with D2ROffline.exe it worked fine !  So i would guess VMs are like Win 7 unsupported ?

**Comment by malvarenga123** [comment]
@Chris-HC1 do you still have the logs?

**Comment by prusswan** [comment]
It is likely blocking or not working on Win 7 (encountered same black screen issue)

**Comment by creambyemute** [comment]
The Alpha itself only supports Windows 10. Look at the required minimum specs!
